packages/react-charts/src/components/ChartUtils/chart-patterns.tsx

```diff
@@ -331,6 +331,9 @@ export const getDefaultPatternProps = ({
   let patternId;
 
   if (isPatternDefs) {
+    // TODO: React hooks cannot be used in regular functions (https://reactjs.org/docs/hooks-rules.html#only-call-hooks-from-react-functions).
+    // This call to 'useMemo()` needs to be replaced with something else, or this function needs to be converted to a React component.
+    // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
I had to disable the linting rule here as making this follow the rules would mean refactoring a public API, which is beyond the scope of this PR.

That said, this is wrong and needs to be fixed.

Reply:
It seems like we can just remove the useMemo call as a solution, but would need to test to be sure. I think we should open a ticket if we don't plan on fixing now so that this TODO does not get forgotten.

Reply:
```
export const getDefaultPatternProps = ({
  colorScale,
  hasPatterns,
  patternScale,
  themeColorScale
}: PatternPropsInterface) => {
  const defaultColorScale = getDefaultColorScale(colorScale, themeColorScale);
  let defaultPatternScale = patternScale;
  let isPatternDefs = !patternScale && hasPatterns !== undefined;
  const patternId = React.useRef<string>();

  if (isPatternDefs) {
    patternId.current = getPatternId();
    defaultPatternScale = getDefaultPatternScale({
      colorScale: defaultColorScale,
      patternId: patternId.current,
      patternScale
    });
  }

  // Initialize pattern visibility
  if (Array.isArray(hasPatterns)) {
    for (let i = 0; i < defaultPatternScale.length; i++) {
      if (!(i < hasPatterns.length && hasPatterns[i])) {
        defaultPatternScale[i] = null;
      }
    }
  } else if (hasPatterns === false) {
    defaultPatternScale = undefined;
    isPatternDefs = false;
  }
  return {
    defaultColorScale,
    defaultPatternScale,
    isPatternDefs,
    patternId: isPatternDefs ? patternId.current : undefined
  };
};
```
This would fix it w/o any refactoring.

Reply:
@jeff-phillips-18 The solution you provided still uses a hook outside of a React component, so that won't work unfortunately.

I don't think that it is possible to remove this memoization as it calls Lodash's [`uniqueId`](https://lodash.com/docs/4.17.15#uniqueId), which makes me assume that this is intentional to preserve the correct id between renders.

I've logged #7909 to resolve this another time, I think for now the impact would be too large.

Reply:
It'd be a hook if it were just named correctly ☹️

Reply:
I could rename it, but I am not quite sure if it's exposed as a public API.

Reply:
Created a PR where we can continue this discussion: #7912